system-tests/management-api/management-api-test-runner/src/test/java/org/eclipse/edc/test/e2e/managementapi/TransferProcessApiEndToEndTest.java

```diff
@@ -243,6 +243,48 @@ void request_byState() throws JsonProcessingException {
             assertThat(result).anySatisfy(it -> assertThat(it.asJsonObject().getString("state")).isEqualTo(state.toString()));
         }
 
+        @Test
+        void request_sortByStateTimestamp() throws JsonProcessingException, InterruptedException {
+            var tp1 = createTransferProcessBuilder("test-tp1").build();
+            var tp2 = createTransferProcessBuilder("test-tp2").build();
+            getStore().save(tp1);
+            Thread.sleep(3000);
+            tp2.updateStateTimestamp();
+            getStore().save(tp2);
+
+
+            var content = """
+                    {
+                        "@context": {
+                            "@vocab": "https://w3id.org/edc/v0.0.1/ns/"
+                        },
+                        "@type": "QuerySpec",
+                        "sortField": "stateTimestamp",
+                        "sortOrder": "ASC",
+                        "limit": 100,
+                        "offset": 0
+                    }
+                    """;
+            var query = JacksonJsonLd.createObjectMapper()
+                    .readValue(content, JsonObject.class);
+
+            var result = baseRequest()
+                    .contentType(JSON)
+                    .body(query)
+                    .post("/v2/transferprocesses/request")
+                    .then()
+                    .log().ifError()
+                    .statusCode(200)
+                    .extract().body().as(JsonArray.class);
+
+            assertThat(result).isNotEmpty().hasSize(2);
+            assertThat(result).isSortedAccordingTo((o1, o2) -> {
+                Long l1 = o1.asJsonObject().getJsonNumber("stateTimestamp").longValue();
+                Long l2 = o2.asJsonObject().getJsonNumber("stateTimestamp").longValue();
```

Comment:
`var`

Reply:
that won't work, bc `var` would infer `long`, but we need `Long` to be able to `compareTo`

Reply:
I think it can be done with `Long.compare(long x, long y)` (got this from the internet, not tried it out)

Reply:
ok, fine :)